source/common/upstream/upstream_impl.h

```diff
@@ -822,5 +822,17 @@ class BaseDynamicClusterImpl : public ClusterImplBase {
  */
 Network::DnsLookupFamily getDnsLookupFamilyFromCluster(const envoy::api::v2::Cluster& cluster);
 
+/**
+ * Utility function to report upstream cx destroy metrics
+ */
+void reportUpstreamCxDestroy(Upstream::HostDescriptionConstSharedPtr host,
```

Comment:
nit: I would pass `const Upstream::HostDescription&` to each of these functions.